### PR TITLE
update MvxWindowPresentation to support creation from Storyboard for Mac

### PR DESCRIPTION
**Pull request by tofutim**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Allows developer to visually create a custom window inside IB (including toolbars, etc.) which will be used to preset the view.

### :arrow_heading_down: What is the current behavior?
Currently a window is created by code and modified using attribute parameters.

### :new: What is the new behavior (if this is a feature change)?
A new attribute for Mac views is introduced. It used the following way:
```
    [MvxFromStoryboard("Main")]
    [StoryboardWindowPresentation("CustomWindow", "Main")]
    public partial class RootView : MvxViewController<RootViewModel>, IMvxOverridePresentationAttribute
	{
        public CustomWindow WindowController {
            get {
                return View.Window != null ? ((CustomWindow)View.Window.WindowController) : null;
            }
        }
```
bindings can be made to outlets in the WindowController using something like
```
            set.Bind(WindowController.labelTitle).For(v => v.StringValue).To(vm => vm.Title);
```

### :boom: Does this PR introduce a breaking change?
Nope.

### :bug: Recommendations for testing
Create a new Mac application and a window with custom features in IB. The new window should hold your base view.

### :memo: Links to relevant issues/docs
https://github.com/MvvmCross/MvvmCross/issues/2180
https://github.com/MvvmCross/MvvmCross/issues/2178

### :thinking: Checklist before submitting

- [x] All projects build
- [x] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [x] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [x] Nuspec files were updated (when applicable) *n/a*
- [x] Rebased onto current develop

### CODE REVIEW

**Comment by tofutim** [comment]
Modified playground sample to use storyboard ToolbarWindow but you may want a separate button. Wasn't sure if I should change the .Core project to add a new view, so I will leave to you....

**Comment by tofutim** [comment]
I think I might fiddle with the sample a bit more and use the WindowViewModel for this.

**Comment by tofutim** [comment]
I made some changes to the sample so that the StoryboardWindow is used when clicking on Show Window. Please let me know if there are any questions, but it should be pretty clear now. The only catch is that binding to the WindowController has to happen at ViewDidAppear because only then is the Window available.

**Comment by tofutim** [comment]
I thought about this. But there is a problem on what settings should take priority. For example the width. Should it take the width from the attribute or the storyboard? I recommend two separate presenters

**Comment by martijn00** [comment]
What we have done on Android is for example putting a default value of int.MinVal in and then check in the presenter if that is used then use the value of the view(storyboard), but if it is set than take that value.

**Comment by tofutim** [comment]
I still think it is a bit odd, what if the storyboard width is 500, the default width is 200, and the user writes in 200 in the code? you would end up with the storyboard width since the default width is 200. The only way I can think of logically doing this is to have a starting width of something like -1. Then if there is no storyboard, it will set the width to 200.

**Comment by tofutim** [comment]
Ironically I have now created a complex method which got flagged

image

**Comment by tofutim** [comment]
Please let me know if the result is what you are looking for, remarkably it works.

File: MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs [hunk 1]
```diff
@@ -6,18 +6,50 @@ namespace MvvmCross.Mac.Views.Presenters.Attributes
 {
     public class MvxWindowPresentationAttribute : MvxBasePresentationAttribute
     {
-        public float PositionX { get; set; } = 200;
+        public static float DefaultPositionX = 200;
```
Comment (martijn00) [inline MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs]: Did you make those properties static so you can set default values from your own code?

**Comment by martijn00** [comment]
Looks great to me! Just one question about the static values.

File: MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs (hunk 1 above)
Comment (tofutim) [reply MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs]: No I wasn't that smart. Just didn't think that we should keep making the same DefaultPositionX for each class. But now that you mention it, good idea!

Comment (tofutim) [reply MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs]: Probably initialpositionx should be const static though

Comment (tofutim) [reply MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs]: Not possible?

image

File: docs/_documentation/presenters/mac-view-presenter.md [hunk 2]
```diff
@@ -59,6 +59,19 @@ The presentation can be customized through this properties:
 - TabTitle: Defines the title of the tab that will be displayed in the segmented control for tabs.
 - WindowIdentifier: identifier for the window where the view should be displayed.
 
+### MvxStoryboardWindowPresentationAttribute
```
Comment (martijn00) [inline docs/_documentation/presenters/mac-view-presenter.md]: I think this documentation needs to be merged into the old attribute now we will keep that.

File: MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs (hunk 1 above)
Comment (martijn00) [reply MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs]: Just leave it static. After the docs are updated i'm all good to merge.

File: docs/_documentation/presenters/mac-view-presenter.md (hunk 2 above)
Comment (tofutim) [reply docs/_documentation/presenters/mac-view-presenter.md]: ah yes good catch

File: MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs (hunk 1 above)
Comment (tofutim) [reply MvvmCross/Mac/Mac/Views/Presenters/Attributes/MvxWindowPresentationAttribute.cs]: I changed the initials to public float {get;}

File: docs/_documentation/presenters/mac-view-presenter.md (hunk 2 above)
Comment (tofutim) [reply docs/_documentation/presenters/mac-view-presenter.md]: i changed the docs

**Pull request merged**